### DISCUSSION TITLE
Xray(VLESS+XHTTP)+Nginx 失败

### DISCUSSION THREAD

**Discussion by juneszh** [lxhao61/integrated-examples#225] [General]
服务器原本在用http upgrade，正常使用。

Xray和Nginx均最新，按照配置说明删掉Caddy和低版本的配置，并替换该替换的变量，测了一天都失败，port和uds都试过。

nginx日志收到请求，应该和xray通信有问题。

nginx/access.log

> 1.2.3.4- - [06/Dec/2024:18:19:30 +0800] "GET /abc/e00a7cfa-c902-4712-8942-81390f625964?x_padding=00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000 HTTP/2.0" 404 513 "-" "Go-http-client/2.0" "1.2.1.2"

nginx/error.log

> 2024/12/06 18:19:30 [error] 5849#5849: *21 connect() to [2606:4700:3037::ac43:c1a4]:443 failed (101: Network is unreachable) while connecting to upstream, client: 1.2.3.4, server: c.abc.cc, request: "GET /abc/3acd1648-5269-4a8a-bfdb-10d78dc2c0c3?x_padding=0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000 HTTP/2.0", upstream: "https://[2606:4700:3037::ac43:c1a4]:443/abc/3acd1648-5269-4a8a-bfdb-10d78dc2c0c3?x_padding=0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000", host: "c.abc.cc"

xray/error.log
> 2024/12/06 18:19:00 [Info] transport/internet/splithttp: failed to validate path, request:/abc, config:/abc/

**Comment by lxhao61** [comment]
请查看 Xray 近期版本升级注意事项，建议服务端与客户端同一个版本，且最好使用最新的。
